blockchain/validation.go

```diff
@@ -69,8 +121,51 @@ func (bc *BlockChain) ValidTransaction(t *Transaction) (bool, TransactionCode) {
 	return true, ValidTransaction
 }
 
+// ValidCloudBase returns true if a transaction is a valid CloudBase transaction
+// and false otherwise
+func ValidCloudBase(t *Transaction) (bool, CloudBaseTransactionCode) {
```

Comment:
There is also potentially a difference between validating a block that we receive from another node and validating a block that we ourselves are mining?